extra/interface/src/main/java/org/spongepowered/configurate/interfaces/meta/range/NumericRange.java

```diff
@@ -0,0 +1,35 @@
+package org.spongepowered.configurate.interfaces.meta.range;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation limits the values that a config node can have.
+ * Because of annotation limits, there is an annotation for:
+ * {@link DecimalRange decimals}, {@link NumericRange numerics} and
+ * {@link StringRange String length}.
+ *
+ * @since 4.2.0
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface NumericRange {
```

Comment:
These range annotations shouldn't be part of the interface extra module -- if we have these sorts of things, they should be constraints in core (and as a separate PR)